heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraCLIController.java

```diff
@@ -111,29 +113,45 @@ public void removeContainers(Set<PackingPlan.ContainerPlan> containersToRemove)
   }
 
   @Override
-  public void addContainers(Integer count) {
+  public Set<Integer> addContainers(Integer count) {
     //aurora job add <cluster>/<role>/<env>/<name>/<instance_id> <count>
     //clone instance 0
     List<String> auroraCmd = new ArrayList<>(Arrays.asList(
-        "aurora", "job", "add", "--wait-until", "RUNNING", jobSpec + "/0", count.toString()));
-
-    if (isVerbose) {
-      auroraCmd.add("--verbose");
-    }
+        "aurora", "job", "add", "--wait-until", "RUNNING",
+        jobSpec + "/0", count.toString(), "--verbose"));
 
     LOG.info(String.format("Requesting %s new aurora containers %s", count, auroraCmd));
-    if (!runProcess(auroraCmd)) {
+    StringBuilder stderr = new StringBuilder();
+    if (!runProcess(auroraCmd, null, stderr)) {
       throw new RuntimeException("Failed to create " + count + " new aurora instances");
     }
+
+    if (stderr.length() <= 0) { // no container was added
+      throw new RuntimeException("empty output by Aurora");
+    }
+    return extractContainerIds(stderr.toString());
+  }
+
+  private Set<Integer> extractContainerIds(String auroraStdoutStr) {
+    String pattern = "Querying instance statuses: [";
+    int idx1 = auroraStdoutStr.indexOf(pattern) + pattern.length();
```

Comment:
I think this line might have a bug.

idx1 is not -1 when the string is not found. Instead, it is -1 + pattern.length(). So you cannot compare it with 0.

Suggest to add a unit test for the function to make sure it works correctly in both successful and failure cases.

Reply:
updated